src/backend/shared/Helpers.ts

```diff
@@ -7,3 +9,45 @@ export function uniqueValidator(uniqueProperties: any, model: any): Promise<bool
         })
     });
 }
+
+// given current user, and list of all users, sort all users from high to low, based on recommended score
+export function applyRecommendation(currentUser: IUser, userList: IUser[]) {
```

Comment:
I still think adding a `score` on the model and the interface would have some side effect... because it changes depending on who the currentUser is. So maybe instead, you can do something like this:

```suggestion
interface UserDictionary {
    [userId: string]: {user: IUser, score: number};
}

export function applyRecommendation(currentUser: IUser, userList: IUser[]) {
    const scoresByUserId = userList.reduce(
        ((acc, comparedUser) => {
            acc[comparedUser._id] = {user: comparedUser, score: calculateRecommendScore(currentUser, comparedUser)};
            return acc;
        }), {} as UserDictionary);

    return Object.entries(scoresByUserId).sort((a, b) => a[1].score > b[1].score ? -1:1).map(val => val[1].user);
}

export function calculateRecommendScore(currentUser: IUser, comparedUser: IUser): number{
    let commonTagNum = findCommonTagsNum(currentUser, comparedUser);
    let resultScore = 0;
    if (currentUser.major && comparedUser.major) {
        if (currentUser.major.toLowerCase() === comparedUser.major.toLowerCase()) {
            resultScore += 10 + commonTagNum * 5;
        } else {
            resultScore += commonTagNum * 5;
        }
   }
    return resultScore;
}
```

Reply:
This is a good comment, I appreciate it, thanks